src/jltypes.c

```diff
@@ -2133,16 +772,9 @@ static jl_value_t *lookup_type_stack(jl_typestack_t *stack, jl_datatype_t *tt, s
 static size_t jl_type_depth(jl_value_t *dt)
 {
     if (jl_is_uniontype(dt)) {
-        jl_svec_t *t = ((jl_uniontype_t*)dt)->types;
-        size_t i, l = jl_svec_len(t);
-        size_t depth = 0;
-        for (i = 0; i < l; i++) {
-            jl_value_t *p = jl_svecref(t, i);
-            size_t d = jl_type_depth(p);
-            if (d > depth)
-                depth = d;
-        }
-        return depth;
+        size_t ad = jl_type_depth(((jl_uniontype_t*)dt)->a);
+        size_t bd = jl_type_depth(((jl_uniontype_t*)dt)->b);
+        return ad > bd ? ad : bd;
     }
     else if (jl_is_datatype(dt)) {
```

Comment:
missing case for `UnionAll` (I think that case is equal to `jl_type_depth(ua->body)`)?